components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -3530,6 +3553,67 @@ public void setGlobalRbacScopeIssuerEnabled(boolean globalRbacScopeIssuerEnabled
         this.globalRbacScopeIssuerEnabled = globalRbacScopeIssuerEnabled;
     }
 
+    public boolean isTlsClientCertificateBoundAccessTokensSupported() {
+
+        return tlsClientCertificateBoundAccessTokensSupported;
+    }
+
+    public List<String> getSupportedTokenEndpointAuthMethods() {
+
+        return supportedTokenEndpointAuthMethods;
+    }
+
+    public List<String> getSupportedTokenEndpointSigningAlgorithms() {
+
+        return supportedTokenEndpointSigningAlgorithms;
+    }
+
+    /**
+     * Parse supported client authentication methods and add them to the supportedTokenEndpointAuthMethods list.
+     *
+     * @param methods OMElement of supported authentication methods.
+     */
+    private void parseSupportedTokenEndpointAuthMethods(OMElement methods) {
+
+        if (methods == null) {
+            return;
+        }
+
+        Iterator iterator = methods.getChildrenWithLocalName(
+                ConfigElements.SUPPORTED_TOKEN_ENDPOINT_AUTH_METHOD);
+        if (iterator != null) {
+            for (; iterator.hasNext(); ) {
+                OMElement method = (OMElement) iterator.next();
+                if (method != null) {
+                    supportedTokenEndpointAuthMethods.add(method.getText());
+                }
+            }
+        }
+    }
+
+    /**
+     * Parse supported signing algorithms and add them to the supportedTokenEndpointSigningAlgorithms list.
+     *
+     * @param algorithms OMElement of supported algorithms.
+     */
+    private void parseSupportedTokenEndpointSigningAlgorithms(OMElement algorithms) {
```

Comment:
Instead of doing the mapped algorithm check in the ProviderConfigBuilder we can do it here itself and maintain the list. This way we can detect any failures early on as well.

Reply:
Fixed in a0e00c2b1fae6d3e771343b8dcc8840d1787da91